applications/ecallmgr/src/ecallmgr_fs_recordings.erl

```diff
@@ -1,228 +1,45 @@
+    'ok'.
 
-%%--------------------------------------------------------------------
-%% @private
-%% @doc
-%% Convert process state when code is changed
-%%
-%% @spec code_change(OldVsn, State, Extra) -> {ok, NewState}
-%% @end
-%%--------------------------------------------------------------------
--spec code_change(any(), state(), any()) -> {'ok', state()}.
-code_change(_OldVsn, State, _Extra) ->
-    {'ok', State}.
-
-%%%===================================================================
-%%% Internal functions
-%%%===================================================================
--spec init_props(kz_proplist(), kz_proplist()) -> kz_proplist().
-init_props(Props, Options) ->
-    case props:get_is_true(<<"Publish-Channel-State">>, Props) of
-        'undefined' ->
-            case props:is_false(<<"Publish-Channel-State">>, Options, 'false') of
-                'true' -> props:set_value(<<"Publish-Channel-State">>, 'false', Props);
-                _ -> Props
-            end;
-        _Value -> Props
-    end.
-
--spec handle_record_event(atom(), kz_proplist(), kz_proplist()) -> 'ok'.
-handle_record_event(Node, [UUID | FSProps], Options) ->
-    kz_util:put_callid(UUID),
-    Props = init_props(FSProps, Options),
-    process_event(UUID, Props, Node),
-    maybe_publish_record_event(Props).
-
--spec process_event(api_binary(), kz_proplist(), atom()) -> any().
-process_event(UUID, Props, Node) ->
+-spec handle_record_stop(map()) -> any().
+handle_record_stop(#{node := Node, call_id := UUID, payload := JObj}) ->
     kz_util:put_callid(UUID),
-    EventName = kzd_freeswitch:event_name(Props),
-    process_specific_event(EventName, UUID, Props, Node).
-
--spec process_specific_event(ne_binary(), api_binary(), kz_proplist(), atom()) -> any().
-process_specific_event(<<"RECORD_STOP">>, UUID, Props, Node) ->
-    IsLocal = props:is_true(<<"Force-Publish-Event-State">>, Props, 'true')
-        orelse (props:is_true(<<"Publish-Channel-State">>, Props, 'true')
-                andalso handling_locally(Props)
-               ),
-    Args = [IsLocal, kzd_freeswitch:media_recorder(Props), Props, UUID, Node],
-    kz_util:spawn(fun maybe_store_recording/5, Args);
-process_specific_event(<<"RECORD_START">>, _UUID, _Props, _Node) -> 'ok';
-process_specific_event(_Event, _UUID, _Props, _Node) ->
-    lager:debug("event ~s for callid ~s not handled in recordings (~s)", [_Event, _UUID, _Node]).
-
-
--spec maybe_publish_record_event(kz_proplist()) -> 'ok'.
-maybe_publish_record_event(Props) ->
-    case props:is_true(<<"Force-Publish-Event-State">>, Props, 'true')
-        orelse (props:is_true(<<"Publish-Channel-State">>, Props, 'true')
-                andalso handling_locally(Props)
-               )
-    of
-        'true' -> ecallmgr_call_events:process_channel_event(Props);
-        'false' -> lager:debug("not publishing record event ~s", [kzd_freeswitch:event_name(Props)])
-    end.
+    IsLocal = handling_locally(JObj),
```

Comment:
This is not being passed a `JObj` but the spec for `handling_locally/1` expects a `kz_proplist()`. Please update that spec